doc/mgr/dashboard.rst

```diff
@@ -298,6 +301,171 @@ previously defined username and password. Select the **Keep me logged in**
+consists of a username, a password (stored in encrypted form using ``bcrypt``),
+an optional name, and an optional email address.
+
+User accounts are stored in MON's configuration database, and are globally
+shared across all ceph-mgr instances.
+
+We provide a set of CLI commands to manage user accounts:
+
+- *Show User(s)*::
+
+  $ ceph dashboard ac-user-show [<username>]
+
+- *Create User*::
+
+  $ ceph dashboard ac-user-create <username> <password> [<rolename>] [<name>] [<email>]
+
+- *Delete User*::
+
+  $ ceph dashboard ac-user-delete <username>
+
+- *Change Password*::
+
+  $ ceph dashboard ac-user-set-password <username> <password>
+
+- *Modify User (name, and email)*::
+
+  $ ceph dashboard ac-user-set-info <username> <name> <email>
+
+
+User Roles and Permissions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+User accounts are also associated with a set of roles that define which
+dashboard fuctionality can be accessed by the user.
+
+Dashboard functionality/modules are grouped within a *security scope*.
+Security scopes are predefined and static. The current avaliable security
+scopes are:
+
+- **hosts**: includes all features related with the ``Hosts`` menu
+  entry.
+- **config-opt**: includes all features related with management of Ceph
+  configuration options.
+- **pool**: includes all features related with pool management.
+- **osd**: includes all features related with OSD management.
+- **monitor**: includes all features related with Monitor management.
+- **rbd-image**: includes all features related with RBD image
+  management.
+- **rbd-mirroring**: includes all features related with RBD-Mirroring
+  management.
+- **iscsi**: includes all features related with iSCSI management.
+- **rgw**: includes all features related with Rados Gateway management.
+- **cephfs**: includes all features related with CephFS management.
+- **manager**: include all features related with Ceph Manager
+  management.
+- **log**: include all features related with Ceph logs management.
+
+A *role* specifies a set of mappings between a *security scope* and a set of
+*permissions*. There are four types of permissions:
+
+- **read**
+- **create**
+- **update**
+- **delete**
+
+Below we show an example of a role specification based on a Python dictionary::
+
+  # example of a role
+  {
+    'role': 'my_new_role',
+    'scopes_permissions': {
+      'pool': ['read', 'create'],
+      'rbd-image': ['read', 'create', 'update', 'delete']
+    }
+  }
+
+The above role dictates that a user has *read* and *create* permissions for
+features related with pool management, and has full permissions for
+features related with RBD image management.
+
+The Dashboard already provides a set of predefined roles that we call
+*system roles*, and can be used right away in a fresh Ceph Dashboard
+installation.
+
+The list of system roles are:
+
+- **Administrator**: provides full permissions for all security scopes.
+- **Read-only**: provides *read* permission for all security scopes.
```

Comment:
The role name is "Read-Only" (capital "O"):
```
# ./bin/ceph dashboard ac-user-add-roles lenz "Read-only"
Error ENOENT: Role 'Read-only' does not exist
# ./bin/ceph dashboard ac-user-add-roles lenz "Read-Only"
{"username": "lenz", "password": "$2b$12$R23X03BLMgzOrmLMYXnXCe4MiO3Oo.3xbVIvO4MAk8OAhHzMCUDHu", "name": null, "roles": ["Pool Manager", "Read-Only"], "email": null}
```
I wonder if the role names should be "less fancy", which makes it somewhat convenient to enter them on the command line, e.g. just consisting of a single word (no blanks), just allowing certain special chars like dash and dot?

Reply:
I'm also wondering if it makes sense to accept roles written in lower-case as well?